src/component/Legend.tsx

```diff
@@ -157,6 +164,12 @@ export class Legend extends PureComponent<Props, State> {
   }
 
   public render() {
-    return <LegendWrapper {...this.props} />;
+    return (
+      <g>
+        <foreignObject x="0" y="0" width="100%" height="100%" style={{ pointerEvents: 'none' }}>
```

Comment:
why does this need a foreignObject? Legend is always an SVG no?

Also why `pointerEvents: 'none'`?

Reply:
Yes so Legend is a `<div>`, same as Tooltip (both the wrapper and content are HTML elements, not SVG).

When I tried run tests without the `foreignObject` I found two problems, one our tooling fails because it only works with HTMLElement but not SVGElement. That's fixable, what's worse is that jsdom throws when recharts starts reading offsetWidth and offsetHeight:

```
TypeError: 'get offsetWidth' called on an object that is not a valid instance of HTMLElement.
    at SVGElement.get offsetWidth [as offsetWidth] (recharts/node_modules/jsdom/lib/jsdom/living/generated/HTMLElement.js:441:15)
    at recharts/src/util/useGetBoundingClientRect.ts:33:23
```

as jsdom is out of our reach, and I am not ready to refactor recharts to not use offsetWidth, I googled and added foreignObject and it all works.

Also without foreignObject the tooltip doesn't display at all in Sankey and other charts that do not have the `renderTooltip` method anymore. I suspect it's because of the offsetWidth problem but I didn't dig deeper into why it doesn't work.

Here's an example where embedded HTML breaks the SVG: https://codesandbox.io/p/sandbox/foreignobject-in-svg-ymvzlx?file=%2Findex.html%3A28%2C40

And this is what it looks like in my Firefox:
<img width="687" alt="image" src="https://github.com/recharts/recharts/assets/1100170/213fabc4-2a89-4816-bb50-48776abdc68f">

Now I suspect that this is the original reason of why both tooltip and legend are rendered outside of `<Surface>` (the `<svg>` root) in all charts. Since we don't want element cloning anymore we're going to have to do with foreignObject.

Why pointer-events: none? When I was fixing the Tooltip I noticed that mouse hover no longer works. I attributed that to the fact that there's now a `foreignObject` element stretched on top of the whole thing and I added pointer-events: none because the blank element is not meant to catch any mouse interaction in the first place and that fixed it. I can't get it to reproduce in codesandbox though.

Reply:
Thank you for the explanation!

We should make sure it's documented somewhere (comments I guess?) why we do this as I'm sure I'll ask the same question again in 2 years 😂